apps/api/src/app/events/events.module.ts

```diff
@@ -36,11 +41,15 @@ import { LayoutsModule } from '../layouts/layouts.module';
   controllers: [EventsController],
   providers: [
     ...USE_CASES,
-    WorkflowQueueProducerService,
+    {
+      provide: QueueService,
+      useClass: WorkflowQueueProducerService,
+    },
```

Comment:
the `QueueService` is used by the `AddJob` use-case which is used by API and `worker-service` apps, so `AddJob` should use the queue interface instead of knowing about the specific implementation, which for the API is `WorkflowQueueProducerService` and for the `worker-service` is `WorkflowQueueService` class with processing logic